plugins/optimization-detective/class-od-visited-tag-state.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Optimization Detective: OD_Visited_Tag_State class
+ *
+ * @package optimization-detective
+ * @since n.e.x.t
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * State for a tag visitation when visited by tag visitors while walking over a document.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+final class OD_Visited_Tag_State {
```

Comment:
I'm all for classes where reasonable, but maybe this one goes a bit too far, since it's only a wrapper for a single boolean value? Some more verbosely architected PHP projects could work with this, but it seems excessive for WordPress.

Is there any reason a class is helpful here? Or could this simply be a `bool` property in `OD_Tag_Visitor_Context`? Since it's that class that's passed around to the callbacks and since it's already for a specific tag, couldn't we put the methods into that class?

If we do that, I think only the `reset()` method wouldn't work well at least per its name, since then that name would be too generic. But we might as well use `track_tag()` as a setter with a boolean parameter, e.g. like `track_tag( $track = true )`.

Reply:
The reason why I did it this way is that the tag visitors shouldn't have access to the underlying state. They shouldn't be able to override whether or not another tag visitor requested to track the tag. So in this way, `od_optimize_template_output_buffer()` instanties the `OD_Visited_Tag_State` object and it has access to its internal state, but then it passes this into `OD_Tag_Visitor_Context` which does not expose any of that state to the tag visitors.

Reply:
Ah I see, that makes sense. Thanks!